File: lambda-kinesis-lambda/producer/producer-function.py
#! Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
#  SPDX-License-Identifier: MIT-0
from concurrent.futures import process
import boto3
import json
from datetime import datetime
import calendar
import random
import time
import os
  
stream_name = os.environ['KINESIS_STREAM']
kinesis_client = boto3.client('kinesis', region_name=os.environ['AWS_REGION'])

def put_to_stream(property_id, property_value, property_timestamp):
    payload = {
                'prop': str(property_value),
                'timestamp': str(property_timestamp),
                'property_id': property_id
              }
    print(payload)
    put_response = kinesis_client.put_record(
                        StreamName=stream_name,
                        Data=json.dumps(payload),
                        PartitionKey=property_id)

def lambda_handler(event, context):
    property_value = random.randint(40, 120)
    property_timestamp = calendar.timegm(datetime.utcnow().timetuple())
    property_id = 'Property11'
    put_to_stream(property_id, property_value, property_timestamp)
    return {"statusCode":200, "body":"Successfully posted"}

File: lambda-kinesis-lambda-js/producer/producer-function.js
// ! Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
//   SPDX-License-Identifier: MIT-0
const { KinesisClient, PutRecordCommand } = require('@aws-sdk/client-kinesis');

const streamName = process.env.KINESIS_STREAM;
const kinesisClient = new KinesisClient();

function putToStream(propertyId, propertyValue, propertyTimestamp) {
    const payload = {
        prop: String(propertyValue),
        timestamp: String(propertyTimestamp),
        property_id: propertyId
    };

    console.log(payload);

    const params = {
        StreamName: streamName,
        Data: JSON.stringify(payload),
        PartitionKey: propertyId
    };
    //return params
    return kinesisClient.send(new PutRecordCommand(params));
}

exports.lambda_handler = async (event, context) => {
    const propertyValue = Math.floor(Math.random() * (120 - 40 + 1)) + 40;
    const propertyTimestamp = Math.floor(new Date().getTime() / 1000);
    const propertyId = 'Property11';

    try {
        await putToStream(propertyId, propertyValue, propertyTimestamp);
        return { statusCode: 200, body: 'Successfully posted' };
    } catch (error) {
        console.error('Error:', error);
        return { statusCode: 500, body: 'Error posting to stream' };
    }
};